tests/Builder/Query/RegexOperatorTest.php

```diff
@@ -18,8 +18,7 @@ public function testPerformALIKEMatch(): void
     {
         $pipeline = new Pipeline(
             Stage::match(
-                // sku: \MongoDB\Builder\Query::regex('789$', ''),
-                sku: new Regex('789$', ''),
+                sku: Query::regex('789$', ''),
```

Comment:
Looks like comparing JSON allows to fix this issue: https://github.com/mongodb/mongo-php-builder/pull/6#discussion_r1366187184